actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -29,14 +29,22 @@ def shutdown
       end
 
       def with_connection(&block) # :nodoc:
-        ActiveRecord::Base.connection_pool.with_connection do |ar_conn|
-          pg_conn = ar_conn.raw_connection
-
-          unless pg_conn.is_a?(PG::Connection)
+        fetch_connection do |conn|
+          unless conn.is_a?(PG::Connection)
             raise "The Active Record database must be PostgreSQL in order to use the PostgreSQL Action Cable storage adapter"
           end
 
-          yield pg_conn
+          yield conn
+        end
+      end
+
+      # This method provides the database connection (an instance of PG::Connection)
+      # which is used for the adapter. By default, a connection is taken from
+      # Active Record's connection pool. Override this method if you wish to
+      # use a different ORM.
+      def fetch_connection
+        ActiveRecord::Base.connection_pool.with_connection do |ar_conn|
```

Comment:
Unfortunately I don't think this will solve @rafaelfranca's [concern](https://github.com/rails/rails/issues/27214#issuecomment-263671833) about removing the AR dependency from ActionCable itself. This file would still need to be modified to have a `require 'active_record'`, but that wouldn't be allowed because the gem itself doesn't depend on it.